pkg/model/cache.go

```diff
@@ -244,13 +245,19 @@ func (p *FromCacheProcess) ProcessDictionary(dictionary Dictionary, out Collecto
 func (p *FromCacheProcess) processDictionary(dictionary Dictionary, out Collector) {
 	key, ok := p.selector.Read(dictionary)
 	if !ok {
+		if p.preserve == "notInCache" {
+			out.Collect(dictionary)
+		}
```

Comment:
Is this a fix for a bug ? 
I understand that when the key is not present in the JSON document, then the line was ignored in the output, i'm quite confused 😖

Reply:
It is a fix indeed.
It actually is the origin of this PR:
After a lino pull, I got a result like
```json
{"fk":{"id":1,"name":"Edgard"},"description":"blablabla"}
{"fk":{"id":2,"name":"Candice"},"description":"blablabla2"}
{"fk":[],"description":"blablabla3"}
```
(the third record had no parent so it returned an empty array for the fiel `fk`)
When I modify `fk.id` with a `fromCache` mask, line 3 is ignored in the output as `fk.id` is not present. However, I want to keep line 3.

However, i see two different cases :
- the key **is present** in the JSON document but the value **is not present** in the cache. Here, `preserve: notInCache` can preserve those lines
- the key **is not present** in the JSON document (exemple above). Maybe it needs another option for preserve, like `preserve: missingKey` 

Am I clearer? If yes, what do you think?

Reply:
I think, if the key is not present in the JSON document, then the mask should not add it (consistency with other masks behavior when a key is missing). But it should definitly not remove the line. Correct behavior: the line is untouched.

So this fix is good, it is a real bug ;)

This being said, I don't see why we would need `preserve: missingKey`.

Reply:
So the correct fix should be : 
```go
key, ok := p.selector.Read(dictionary)
if !ok {
	out.Collect(dictionary)
        return
}
```

So the preserve Option `notInCache` does not respond to my problem of missing key, but I still think it could be useful for further purposes.

Reply:
The option respond exactly to the case described in #83 

```
cat data.jsonl | pimo --load-cache mycache=cache.jsonl    
{"id":11,"name":"Rolande"}
{"id":2, "name":"Aaron"}
{"id":13,"name":"Matéo"}
```

For the other case with the missing key, you should consider doing a `add` mask just before.

Reply:
It is useful yes